src/Worker/GrpcInvoker.php

```diff
@@ -79,16 +79,30 @@ public function invokeFinalize(): void
         }
 
         if ($this->kernel instanceof RebootableInterface && $this->dependencies->getKernelRebootStrategy()->shouldReboot()) {
+            try {
+                if ($this->kernel->getContainer()->has('services_resetter')) {
+                    /** @var ResetInterface $resetter */
+                    $resetter = $this->kernel->getContainer()->get('services_resetter');
+                    $resetter->reset();
+                }
+            } catch (\Throwable $e) {
+                $this->logger->error(
+                    \sprintf(
+                        'An error occurred when resetting services: %s',
+                        $e->getMessage()
+                    ),
+                    ['exception' => $e]
+                );
+
+                throw $e;
```

Comment:
I suggest not to rethrow. We restart for a reason. The reason could be because we know the current `Container` is invalid. So we should not handle it twice. 

I think a log message is enough

Reply:
sure, removed throw